.github/workflows/update-gcs-bucket.yaml

```diff
@@ -9,39 +9,62 @@ on:
 
 env:
   node_version: 16
+  typesense_url: https://typesense.camino.network
+  default_collection_name: camino-docs-default
 
 jobs:
-  BuildAndDeploy:
+  build_deploy_scrape:
+    name: Build, Deploy & Scrape
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
-      - uses: actions/setup-node@v3
-        with:
-          node-version: ${{ env.node_version }}
-          
-      - name: install yarn dependencies        
-        run: yarn install
 
-      - name: build
-        run: yarn build
-
-      - name: set bucket names
-        id: setBucketNames
+      - name: Prepare variables
+        id: prepareVariables
         run: |
           if [[ "${{ github.ref_name }}" == "main" ]]; then
               echo "bucket=docs.camino.network" >> $GITHUB_OUTPUT
+              echo "collection=camino-docs-`date +%s`" >> $GITHUB_OUTPUT
+              echo "loadbalancer=docs-lb" >> $GITHUB_OUTPUT
           elif [[ "${{ github.ref_name }}" == "dev" ]]; then
               echo "bucket=dev.docs.camino.network" >> $GITHUB_OUTPUT
+              echo "collection=camino-docs-dev-`date +%s`" >> $GITHUB_OUTPUT
+              echo "loadbalancer=dev-docs-lb" >> $GITHUB_OUTPUT
           fi
 
-      - name: 'gcloud authentication'
-        id: 'auth'
+      - uses: actions/setup-node@v3
+        with:
+          node-version: ${{ env.node_version }}
+          
+      - name: Install yarn dependencies        
+        run: yarn install
+
+      - name: Set collection name for Docusaurus
+        run: sed -i "s/camino-docs-default/${{ steps.prepareVariables.outputs.collection }}/g" docusaurus.config.js
+
+      - name: Build
+        run: yarn build
+
+      - name: Cloud authentication
+        id: auth
         uses: 'google-github-actions/auth@v1'
         with:
           credentials_json: '${{ secrets.GOOGLE_CREDENTIALS }}'
 
-      - name: 'Set up Cloud SDK'
-        uses: 'google-github-actions/setup-gcloud@v1'
+      - name: Set up Cloud SDK
+        uses: google-github-actions/setup-gcloud@v1
+
+      - name: Update bucket
+        run: |
+          gsutil -m rsync -R -d build gs://${{ steps.prepareVariables.outputs.bucket }}
 
```

Comment:
for deploying things please add an extra condition if this is run on dev or main only, because otherwise it will give an error (no bucket name for example if someone runs it with workflow dispatch on x branch) or we can remove the workflow_dispatch trigger.

eg:
if: ${{ github.ref == 'refs/heads/main' || github.ref == 'refs/heads/dev' }}

Reply:
Ah, I was thinking about this. I believe it's better to fail the action if it's not match a branch we want. What do you think?

Reply:
Either remove WD, or check for it IMO

Reply:
Done & tested